monkey/monkey_island/cc/resources/monkey.py

```diff
@@ -66,97 +67,104 @@ def patch(self, guid):
+                    )
+
+            if not db_monkey:
+                monkey_json["parent"] = [parent_to_add]
+            else:
+                monkey_json["parent"] = db_monkey.get("parent") + [parent_to_add]
 
-        if not db_monkey:
-            monkey_json["parent"] = [parent_to_add]
-        else:
-            monkey_json["parent"] = db_monkey.get("parent") + [parent_to_add]
+            tunnel_host_ip = None
+            if "tunnel" in monkey_json:
+                tunnel_host_ip = monkey_json["tunnel"].split(":")[-2].replace("//", "")
+                monkey_json.pop("tunnel")
 
-        tunnel_host_ip = None
-        if "tunnel" in monkey_json:
-            tunnel_host_ip = monkey_json["tunnel"].split(":")[-2].replace("//", "")
-            monkey_json.pop("tunnel")
+            ttl = create_monkey_ttl_document(DEFAULT_MONKEY_TTL_EXPIRY_DURATION_IN_SECONDS)
+            monkey_json["ttl_ref"] = ttl.id
 
-        ttl = create_monkey_ttl_document(DEFAULT_MONKEY_TTL_EXPIRY_DURATION_IN_SECONDS)
-        monkey_json["ttl_ref"] = ttl.id
-
-        mongo.db.monkey.update({"guid": monkey_json["guid"]}, {"$set": monkey_json}, upsert=True)
+            mongo.db.monkey.update(
+                {"guid": monkey_json["guid"]}, {"$set": monkey_json}, upsert=True
+            )
 
-        # Merge existing scanned node with new monkey
+            # Merge existing scanned node with new monkey
 
-        new_monkey_id = mongo.db.monkey.find_one({"guid": monkey_json["guid"]})["_id"]
+            new_monkey_id = mongo.db.monkey.find_one({"guid": monkey_json["guid"]})["_id"]
 
-        if tunnel_host_ip is not None:
-            NodeService.set_monkey_tunnel(new_monkey_id, tunnel_host_ip)
+            if tunnel_host_ip is not None:
+                NodeService.set_monkey_tunnel(new_monkey_id, tunnel_host_ip)
 
-        existing_node = mongo.db.node.find_one(
-            {"ip_addresses": {"$in": monkey_json["ip_addresses"]}}
-        )
+            existing_node = mongo.db.node.find_one(
+                {"ip_addresses": {"$in": monkey_json["ip_addresses"]}}
+            )
 
-        if existing_node:
-            node_id = existing_node["_id"]
-            EdgeService.update_all_dst_nodes(old_dst_node_id=node_id, new_dst_node_id=new_monkey_id)
-            for creds in existing_node["creds"]:
-                NodeService.add_credentials_to_monkey(new_monkey_id, creds)
-            mongo.db.node.remove({"_id": node_id})
+            if existing_node:
+                node_id = existing_node["_id"]
+                EdgeService.update_all_dst_nodes(
+                    old_dst_node_id=node_id, new_dst_node_id=new_monkey_id
+                )
+                for creds in existing_node["creds"]:
+                    NodeService.add_credentials_to_monkey(new_monkey_id, creds)
+                mongo.db.node.remove({"_id": node_id})
 
-        return {"id": new_monkey_id}
+            return {"id": new_monkey_id}
```

Comment:
Check spaces vs. tabs on indents. I'm not sure if github is trying to tell me that some tabs snuck in or just that the indentation changed.

Reply:
I've wrapped this method with `with AGENT_KILLING_SEMAPHORE:` and for that I had to indent the whole method, which is +100 lines of code. That's why it's so unreadable, but the only line with non-indentation changes is `with AGENT_KILLING_SEMAPHORE`. We shouldn't have such a long method, but not sure now is the time to refactor